File: my-folder/0011-container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        L = 0 
        R = len(height) - 1
        max_area = float("-inf")
        while L < R :
            max_area = max(max_area, min(height[L], height[R]) * (R - L))
            if height[R] < height[L]:
                R -= 1
            else: L+=1
           
    
        return max_area

        

File: my-folder/0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] height) {
        int maxWater = 0;
        int left = 0;
        int right = height.length - 1;
        while(left <= right){
            int mini = Math.min(height[left], height[right]);
            maxWater = Math.max(maxWater, mini * (right - left));
            if(height[left] > height[right]){
                right--;
            }
            else{
                left++;
            }
        }

        return maxWater;
    }
}
